docs/getting-started.md

```diff
@@ -1,372 +1,271 @@
-can use the `--ssh-keys` flag to specify the paths to your
-public and private SSH keys on the host. Both of them are
-required for git authentication when fetching the repositories.
+In a new terminal (do not close the 1st one):
 
 ```bash
-tools/devtool build --ssh-keys ~/.ssh/id_rsa.pub ~/.ssh/id_rsa
+TAP_DEV="tap0"
+TAP_IP="172.16.0.1"
+MASK_SHORT="/30"
+
+# Setup network interface
+sudo ip link del "$TAP_DEV" 2> /dev/null || true
+sudo ip tuntap add dev "$TAP_DEV" mode tap
+sudo ip addr add "${TAP_IP}${MASK_SHORT}" dev "$TAP_DEV"
+sudo ip link set dev "$TAP_DEV" up
+
+# Enable ip forwarding
+sudo sh -c "echo 1 > /proc/sys/net/ipv4/ip_forward"
+
+# Set up microVM internet access
+sudo iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE || true
+sudo iptables -D FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT \
+    || true
+sudo iptables -D FORWARD -i tap0 -o eth0 -j ACCEPT || true
+sudo iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
+sudo iptables -I FORWARD 1 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
+sudo iptables -I FORWARD 1 -i tap0 -o eth0 -j ACCEPT
+
+API_SOCKET="./firecracker.socket"
+LOGFILE="./firecracker.log"
+
+# Create log file
+touch $LOGFILE
+
+# Set log file
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"log_path\": \"${LOGFILE}\",
+        \"level\": \"Debug\",
+        \"show_level\": true,
+        \"show_log_origin\": true
+    }" \
+    "http://localhost/logger"
+
+KERNEL="./vmlinux.bin"
+KERNEL_BOOT_ARGS="console=ttyS0 reboot=k panic=1 ipv6.disable=1 i8042.nokbd
+ 8250.nr_uarts=1 random.trust_cpu=on"
+
+# Set boot source
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"kernel_image_path\": \"${KERNEL}\",
+        \"boot_args\": \"${KERNEL_BOOT_ARGS}\"
+    }" \
+    "http://localhost/boot-source"
+
+ROOTFS="./ubuntu-18.04.ext4"
+
+# Set rootfs
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"drive_id\": \"rootfs\",
+        \"path_on_host\": \"${ROOTFS}\",
+        \"is_root_device\": true,
+        \"is_read_only\": false
+    }" \
+    "http://localhost/drives/rootfs"
+
+FC_MAC="06:00:AC:10:00:02"
+
+# Set network interface
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"iface_id\": \"net1\",
+        \"guest_mac\": \"$FC_MAC\",
+        \"host_dev_name\": \"$TAP_DEV\"
+    }" \
+    "http://localhost/network-interfaces/net1"
+
+# API requests are handled asynchronously, it is important the configuration is
+# set, before `InstanceStart`.
+sleep 0.015s
+
+# Start microVM
+curl -X PUT --unix-socket "${API_SOCKET}" \
+    --data "{
+        \"action_type\": \"InstanceStart\"
+    }" \
+    "http://localhost/actions"
+
+# API requests are handled asynchronously, it is important the microVM has been
+# started before we attempt to SSH into it.
+sleep 0.015s
+
+# SSH into the microVM
+sudo ssh -i ./ubuntu-18.04.id_rsa 172.16.0.2
+
+# Use `root` for both the login and password.
+# Run `reboot` to exit.
```

Comment:
Although this includes instructions for net configuration on host, but does not include instructions for net configuration inside guests. How about adding it or redirect to https://github.com/firecracker-microvm/firecracker/blob/main/docs/network-setup.md ?

Reply:
Ah sorry, this seems to be caused by the conflict of ip address range for host and guest.
```
$ ip route
default via 172.31.0.1 dev eth0
169.254.169.254 dev eth0
172.16.0.0/30 dev tap0 proto kernel scope link src 172.16.0.1
172.17.0.0/16 dev docker0 proto kernel scope link src 172.17.0.1 linkdown
172.31.0.0/20 dev eth0 proto kernel scope link src 172.31.11.151
```

Reply:
The guest decides its ip address based on MAC address with [fcnet-setup.sh](https://github.com/firecracker-microvm/firecracker/blob/main/resources/tests/fcnet-setup.sh).

Like my case, in some customers' environment, address range conflicts can happen.
To make `TAP_IP` modification acceptable, it would be nice that `FC_MAC` is derived from `TAP_IP` by doing reverse calculation of `fcnet-setup.sh`.

Reply:
Here is an example.
```
FC_MAC=$(printf "06:00:%02x:%02x:%02x:%02x" $(echo $TAP_IP | sed "s/\./ /g"))
```

Reply:
Would replacing `FC_MAC` with:
```bash
# Derive FC_MAC from TAP_IP e.g. `172.16.0.1` becomes `06:00:AC:10:00:02`.
FC_MAC=$(printf "06:00:%02x:%02x:%02x:%02x" $(echo $TAP_IP | sed "s/\./ /g"))
```
fit this?

Reply:
Included comment on this:
```bash
# The IP address of a guest is derived from its MAC address with
# `fcnet-setup.sh`, this has been pre-configured in the guest rootfs. It is
# important that `TAP_IP` and `FC_MAC` match this.
FC_MAC="06:00:AC:10:00:02"
```